src/NHibernate/Linq/IntermediateHqlTree.cs

```diff
@@ -110,7 +110,7 @@ public void AddDistinctRootOperator()
 			if (!_hasDistinctRootOperator)
 			{
 				Expression<Func<IEnumerable<object>, IList>> x =
-					l => new DistinctRootEntityResultTransformer().TransformList(l.ToList());
+					l => DistinctRootEntityResultTransformer.Instance.TransformList(l.ToList());
```

Comment:
I also want to get rid of this ToList conversion. Will undo this change and make it in my other PR